sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -115,12 +115,6 @@ private[hive] class HiveClientImpl(
   private val outputBuffer = new CircularBuffer()
 
   private val shim = version match {
-    case hive.v12 => new Shim_v0_12()
-    case hive.v13 => new Shim_v0_13()
-    case hive.v14 => new Shim_v0_14()
-    case hive.v1_0 => new Shim_v1_0()
-    case hive.v1_1 => new Shim_v1_1()
-    case hive.v1_2 => new Shim_v1_2()
```

Comment:
I didn't remove `Shim_v0_12` ~ `Shim_v1_2` instances themselves because they inherit each other. I could combine them to one but I think it's fine to leave it as is (as it will show the history at least, and can be useful).

Reply:
those code become unreachable now, I'm not sure it's a good idea to retain the dead code.

even though some users may use a modified low-version hive client that supports JDK 17+, changes like https://github.com/apache/spark/pull/42599 would break them silently.

Reply:
Lemme explicitly throw an exception